sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -163,6 +175,22 @@ private[sql] class JacksonGenerator(
           s"with the type of $dataType to JSON.")
   }
 
+  private def makeReader(dataType: DataType): ValueReader = dataType match {
+
+    case TimestampType =>
+      (row: SpecializedGetters, ordinal: Int) =>
+        options.timestampFormat.format(DateTimeUtils.toJavaTimestamp(row.getLong(ordinal)))
+
+    case DateType =>
+      (row: SpecializedGetters, ordinal: Int) =>
+        options.dateFormat.format(DateTimeUtils.toJavaDate(row.getInt(ordinal)))
+
+    case _ =>
+      (row: SpecializedGetters, ordinal: Int) =>
+        row.get(ordinal, dataType).toString
```

Comment:
Can you check other types and nested types too? I noticed this problem a long time ago but couldn't have some time to fix it. This probably will need to add some tests and check if each type works.

Reply:
I considered this the default case and thought it would be covered by existing Unit Tests.  Which types should I add tests for?  Int, String, Long, etc.  All of the children of AtomicType?

Nested types is interesting.  I will add some tests for keys with a Seq[Int], Map[String, Int] to see that they work.  If there was something else you were thinking of please let me know.

Reply:
Actually, now that I consider it a bit more I think nested types are not possible.  The result has to be stored as a String, e.g.

"key": "value"

Putting the nested type inside the name of the JSON wouldn't be possible.